src/main/java/org/folio/service/PasswordResetLinkServiceImpl.java

```diff
@@ -121,91 +112,59 @@ public Future<String> sendPasswordResetLink(String userId, Map<String, String> o
       .withEventConfigName(eventConfigName)
       .withRecipientId(user.getId())
       .withContext(
         new Context()
           .withAdditionalProperty("user", JsonObject.mapFrom(user))
           .withAdditionalProperty("link", generatedLink)
-          .withAdditionalProperty("expirationTime", expirationTimeFromConfig)
-          .withAdditionalProperty("expirationUnitOfTime", expirationUnitOfTimeFromConfig))
+          .withAdditionalProperty("forgotPasswordLink", forgotPasswordLink))
       .withText(StringUtils.EMPTY)
       .withLang(DEFAULT_NOTIFICATION_LANG);
     return notificationClient.sendNotification(notification, connectionParams);
   }
 
-  private Future<String> signToken(OkapiConnectionParams connectionParams, Boolean passwordExists, Holder<Boolean> passwordExistsHolder, Holder<String> passwordResetActionIdHolder) {
-    passwordExistsHolder.value = passwordExists;
+  private Future<String> signToken(OkapiConnectionParams connectionParams, Holder<String> passwordResetActionIdHolder) {
+    passwordResetActionIdHolder.value = UUID.randomUUID().toString();
+
     JsonObject tokenPayload = new JsonObject()
       .put("sub", UNDEFINED_USER_NAME + passwordResetActionIdHolder.value)
       .put("dummy", true)
       .put("extra_permissions", new JsonArray()
         .add("users-bl.password-reset-link.validate")
         .add("users-bl.password-reset-link.reset")
       );
+
     return authTokenClient.signToken(tokenPayload, connectionParams);
   }
 
-  private Future<Boolean> isPasswordExists(String userId, OkapiConnectionParams connectionParams, Holder<Map<String, String>> configMapHolder, Holder<String> passwordResetActionIdHolder) {
-    String expirationTimeFromConfig =
-      configMapHolder.value.getOrDefault(LINK_EXPIRATION_TIME_CONFIG_KEY, LINK_EXPIRATION_TIME_DEFAULT);
-    String expirationUnitOfTimeFromConfig = configMapHolder.value.getOrDefault(
-      LINK_EXPIRATION_UNIT_OF_TIME_CONFIG_KEY, LINK_EXPIRATION_UNIT_OF_TIME_DEFAULT);
-
-    long expirationTime = convertDateToMilliseconds(expirationTimeFromConfig, expirationUnitOfTimeFromConfig);
-    if (expirationTime > MAXIMUM_EXPIRATION_TIME) {
-      expirationTime = MAXIMUM_EXPIRATION_TIME;
-      configMapHolder.value.put(LINK_EXPIRATION_TIME_CONFIG_KEY, String.valueOf(MAXIMUM_EXPIRATION_TIME_IN_WEEKS));
-      configMapHolder.value.put(LINK_EXPIRATION_UNIT_OF_TIME_CONFIG_KEY, ExpirationTimeUnit.WEEKS.name().toLowerCase());
-    }
+  public Future<Boolean> isPasswordExists(String userId, String token, Holder<String> tokenHolder, OkapiConnectionParams connectionParams, Holder<String> passwordResetActionIdHolder) {
 
-    passwordResetActionIdHolder.value = UUID.randomUUID().toString();
+    tokenHolder.value = token;
+    JsonObject payload = new JsonObject(Buffer.buffer(Base64.getDecoder().decode(token.split("\\.")[1])));
+    Long exp = payload.getLong("exp");
+    Date expirationDate = new Date(exp * 1000);
```

Comment:
What is this 1000 value ? Can this be also stated as a constant ?

Reply:
here we are converting seconds to milliseconds which is required for date object, probably constant is not required for this